src/insertItem.ts

```diff
@@ -152,18 +152,63 @@ export class InsertItem {
         };
         let defaultValue = await this.ui.showInputBox({ prompt: "Default value? Leave empty for no default value.", });
         if (defaultValue) {
-            parameter.defaultValue = defaultValue;
+            parameter.defaultValue = this.getDefaultValue(defaultValue, parameterType.value, context);
         }
         let description = await this.ui.showInputBox({ prompt: "Description? Leave empty for no description.", });
         if (description) {
             parameter.metadata = {
                 description: description
             };
         }
-        await this.insertInObject(template, textEditor, templateKeys.parameters, parameter, name, context);
+        await this.insertInObject(template, textEditor, templateKeys.parameters, parameter, name, context, false);
     }
 
-    private async insertInObject(template: DeploymentTemplate, textEditor: vscode.TextEditor, part: string, data: Data | unknown, name: string, context: IActionContext): Promise<void> {
+    private throwError(message: string, context: IActionContext, suppressReportIssue: boolean = true): void {
+        context.errorHandling.suppressReportIssue = true;
+        throw new Error(message);
+    }
+
+    private getDefaultValue(defaultValue: string, parameterType: string, context: IActionContext): string | number | boolean | undefined {
```

Comment:
##Add
type ItemType  =  "string" | "securestring"|  "int"| "bool" | "bool"|"object"|"secureObject"|"array";

##Change
export function getItemType**s**(): QuickPickItem<**ItemType**>[] {
    let items: QuickPickItem<**ItemType**>[] = [];
    items.push(new QuickPickItem<**ItemType**>("String", "string", "A string"));
etc
...

private getDefaultValue(defaultValue: string, parameterType: **ItemType**, context: IActionContext): string | number | boolean | undefined | {} | [] {
...
switch (parameterType) {
=>
explicitly handle all cases, then use assertNever(parameterType) on default, so that you get a compile-time error if a case is added to ItemType but not handled here.

interface Parameter extends Data {
    // tslint:disable-next-line:no-reserved-keywords
    type: string;
    defaultValue?: string | number | boolean | undefined | {} | [];    ** or you could use just unknown I think **

Reply:
Fixed.